.github/workflows/CI-audit.yml

```diff
@@ -8,32 +8,61 @@ on:
   workflow_call:
   workflow_dispatch:
 
-env:
-  IMAGE_NAME: zencash/sc-ci-base
-  IMAGE_TAG: noble_rust-stable_latest
-  DOCKER_COMPOSE_CMD: "docker compose -f ${GITHUB_WORKSPACE}/ci/docker-compose.yml run --rm cargo-container"
-  DOCKER_BUILD_DIR: /build
-  DOCKER_CARGO_HOME: /tmp/.cargo
-  CARGO_BINARIES_INSTALL: "cargo-audit"
-
 jobs:
   audit:
-    runs-on: ubuntu-latest
+    runs-on: warp-ubuntu-latest-x64-2x
+    name: Cargo audit
     steps:
       - name: Git checkout
         uses: actions/checkout@v4
 
       - name: Cargo audit
-        shell: bash
-        run: ${{ env.DOCKER_COMPOSE_CMD }} cargo audit 2>&1 | tee audit_output.txt
+        uses: ./.github/actions/cargo-cmd
+        with:
+          command: "cargo audit 2>&1 | tee audit_output.txt"
+          use_cache: 'no'
+          cargo_binaries_install: cargo-audit
 
-      - name: Upload output
-        if: ${{ !env.ACT }}
+      - name: Upload cargo audit output
         uses: actions/upload-artifact@v4
         with:
           name: audit-output
           path: audit_output.txt
-          if-no-files-found: warn
-          retention-days: 1
+          retention-days: 7
           compression-level: 0
           overwrite: true
+
+  # Slack notification if failed on scheduled run
+  failure-notification:
+    runs-on: warp-ubuntu-latest-x64-2x
+    name: Notify if failed
+    needs: [audit]
+    if: ${{ failure() && !cancelled() && github.event_name == 'schedule' }}
+    steps:
+      - name: Send Slack notification
+        uses: slackapi/slack-github-action@v1.25.0
+        with:
+          payload: |
```

Comment:
What about to do our own slack action that use the generic one but with just 2 arguments: `header` and `text`?

Reply:
I' don't know if it is possible to have an array for `text` in order to cover the multiple sections scenario.

Reply:
That is a good idea, but to be honest with you IF we go this path, we should deploy it as an action that can be shared by ALL our repos, therefore thinking on creating centralized GH Actions repo and then using those in our workflows since we are using this `SLACK-github-action` under each of the pipeline we wrote.

I would leave this improvement for future task since we probably want to think on how this should be implemented :shrug: